### DISCUSSION TITLE
Code Style

### DISCUSSION THREAD

**Discussion by temikfart** [CoolStory-Studio/coolstory-gram#3] [Техническое]
Надо решить каким кодстайлом будем писать код. Предлагаю за основу как всегда брать [гугловский код стайл](https://google.github.io/styleguide/cppguide.html), но с некоторыми изменениями, которые я считаю необходимыми. Например:
- Я считаю, что к типу относится `&` и `*`, поэтому считаю, что они должны писаться без пробела от названия типа, и с пробелом от идентификатора функции, переменной...

Предложения пишите в комментарии. Когда сформируем общий список, тогда сделаем [Announcement](https://github.com/CoolStory-Studio/coolstory-gram/discussions/2).

**Comment by cann3v** [comment]
Предлагаю между телом функции и `return` оставлять пустую строку.

**Reply by temikfart** [reply]
Думаешь это необходимо? Мне кажется в коротких функциях всё достаточно ясно смотрится:

```C++
int foo(int a, int b) {
    std::vector<int> v(a, b);
    int t = bar(v);
    return pow(t, 2);
}
```

А что делать вот с такими?
```C++
int fizz(int a, int b) {
    if (a >= b) {
        std::vector<int> v(a, b);
        return buzz(v);
    }
    std::vector<int> v(b, a - 1);
    return fizzbuzz(v);
}
```

**Reply by cann3v** [reply]
Да это привычка со времен питона, всегда так делал. В любом случае, ведь пустыми строками отделяются логические блоки кода? Значит когда действительно необходимо, думаю, можно использовать.

**Reply by temikfart** [reply]
Ну в кодстайле об этом ничего не написано, насколько я понял, поэтому по ситуации я думаю будет нормально.

**Comment by c71n93** [comment]
@temikfart @cadnev Это конечно не про кодстайл, но предлагаю добавить обязательную документацию кода.

**Reply by cann3v** [reply]
Комментариями или ты про что?

**Reply by c71n93** [reply]
> Комментариями или ты про что?

да, комментариями

**Reply by cann3v** [reply]
Так точно, командир. Обязательная документация будет.

**Reply by temikfart** [reply]
> Комментариями или ты про что?

Он про doxygen. Почитай про стиль комментариев.
